### ISSUE TITLE
Real hardware test

### ISSUE THREAD

**Issue by wstevens0n** [maxotaku11niku/98VIDEOP#1]
Confirmed working on Pc-9821 Nx/c7 Pentium 100mhz DX4 38mb ram using a CF card.

I've tested with only angelmiku.98v.
There's no sound, just constant static.

There seemed to be no playback delays due to the hardware just lots of sample rate dithering.
![20230324_235622](https://user-images.githubusercontent.com/25314348/227698140-85c19a45-6e75-42a8-b19d-62e3e989e009.jpg)

![20230324_235752](https://user-images.githubusercontent.com/25314348/227698427-08468b98-a437-4e19-869c-a07c8d9be997.jpg)

![20230324_235541](https://user-images.githubusercontent.com/25314348/227698445-ac44b880-b69a-402b-be8b-6e80ef825b96.jpg)

However when the video stopped it did end with a protection fault error.
![20230324_235959](https://user-images.githubusercontent.com/25314348/227698162-4e8a0296-999f-497e-b4cb-930e71052455.jpg)

Great job on this! 🤘

**Comment by maxotaku11niku** [comment]
Wow, great to see that something did work after all! I've only tested this program on emulators so I was expecting something to maybe go wrong on real hardware.
I've just looked up this model, and it appears that it has an 86-compatible soundboard, though not the 86 itself. It might be falling back on buzzer PWM sound, in which case I can see how it would become 'constant static'. Does it even sound vaguely like the original music? Either I went too far with the sample rate (I used 22050 Hz), made some bad assumptions about how 86 PCM works or about how buzzer PWM sound works. I will try my best to see what the problem is even without owning a piece of PC-98 hardware myself. This might be a little difficult.

**Comment by maxotaku11niku** [comment]
The general protection fault likely results from the fact that I use a segment for putting raw file data rather unsafely, since I assume that all conventional memory above where the program is loaded is free for the taking. I never saw this for some reason.

**Comment by maxotaku11niku** [comment]
Oh I just noticed that the error message says that DX = 3FDB which just so happens to be the I/O port number for timer number 1 read/write. I consider this highly unlikely to happen by chance, so I now know that it did fall back on buzzer PWM audio and the general protection fault may come about because the interrupt routine to handle PWM audio is maybe not properly turned off at the end (even though it should be). But I am pretty sure any errors lie with the handling of buzzer PWM audio.

**Comment by maxotaku11niku** [comment]
Ah, I suspect that a laptop speaker doesn't low-pass filter a buzzer PWM signal, so the high frequency components of the raw signal end up being much more dominant than the actual sound. Whether or not I can do anything about this depends on whether you can hear anything recognisable at all, or if it's just noise.

**Comment by maxotaku11niku** [comment]
Perhaps I will find out why it falls back on buzzer PWM audio despite the presence of an 86-compatible soundboard.

**Comment by maxotaku11niku** [comment]
Ah it seems only the FM sound source is 86-compatible, the built-in PCM in your model is not 86-compatible. To support that would require some work, but the main problem is that Neko Project 21 doesn't emulate the right soundchip, so that would make testing difficult.

**Comment by wstevens0n** [comment]
Correct, my model uses MateXPCM (118/wss/canbe) it's different than 86PCM.

However there may be a method to mimic 86PCM. Is there a specific port I/O that's recommended?

Also, you may consider joining the PC98 discord if you're not already. 
https://discord.gg/8DpqC7mD

It would be a great place to ask questions if you ever decide to buy a PC98 or even get assistance with some of your development. 🤘

**Comment by maxotaku11niku** [comment]
It has been several months but I have implemented preliminary support for WSS-compatible sound. I am not certain that it works on real hardware since it works perfectly fine on Neko Project 21, but in a slightly different version of Neko Project it has unacceptable artifacting (alternates between OK sound and being white noise on a regular basis). Also I'm not quite sure if I'm using DMA correctly as well.

**Comment by wstevens0n** [comment]
@maxotaku11niku hello! I would love to go test the newest version for you tonight.

When it comes to WSS usually DMA is either set to 1 or 3.

**Comment by maxotaku11niku** [comment]
Thanks for the reassurance (I used channel 3), I will wait eagerly for the result.
